include/rdma/fi_domain.h

```diff
@@ -128,6 +128,7 @@ enum fi_hmem_iface {
 	FI_HMEM_ZE,
 	FI_HMEM_NEURON,
 	FI_HMEM_SYNAPSEAI,
+	FI_NUM_HMEM_TYPES, /* Keep this flag last to calculate the length of hmem_ops*/
```

Comment:
There's an `OFI_HMEM_MAX` in [include/ofi_mr.h](https://github.com/ofiwg/libfabric/blob/55e77cda043849f47dcb9780e947b4b28fd5d4de/include/ofi_mr.h#L308), though it's hardcoded.

Your addition is better but I'd prefer the name `FI_HMEM_MAX`

Reply:
Please do not modify the application headers.  In particular, you cannot safely add anything like FI_HMEM_MAX,  It does not work.  Once an enum value is defined, regardless of what you call it, *it cannot change* without the potential for breaking an application.

Reply:
Thats a good point.  I couldn't do sizeof(hmem_ops) b/c I initially had this loop in a different file.  I think I can do this now and it will work.

Reply:
I mean FI_HMEM_MAX to be used as an internal sentinel (not used by applications)... in which case I don't think we would break them, but the current impl of not touching the .h is much cleaner.